packages/db-leveldb/README.md

```diff
@@ -0,0 +1,286 @@
+<div id="top"></div>
+
+<br />
+<div align="center">
+
+<h3 align="center">TruffleDB</h3>
+
+  <p align="center">
+    Organize your Blockchain projects before, during, and after deployment
+    <br />
+    <a href="#examples">View Examples</a>
+    ·
+    <a href="https://github.com/trufflesuite/truffle/issues">Report Bug</a>
+    ·
+    <a href="https://github.com/trufflesuite/truffle/issues">Request Feature</a>
+  </p>
+</div>
+
+<!-- TABLE OF CONTENTS -->
+
+  <summary>Table of Contents</summary>
+
+- [Usage](#usage)
+- [Model](#model)
+  - [Model API](#model-api)
+- [Model Instance](model-instance)
+  - [Model Instance API](#model-instance-api)
+- [Examples](#examples)
+  - [GenerateID](#generateid)
+- [Roadmap](#roadmap)
+
+## Usage
+
+```sh
+$ yarn
+
+$ yarn test
+
+$ yarn nyc
+```
+
+```javascript
+const TruffleDB = require("@truffle/db");
+const db = new TruffleDB();
+const { Project } = db.models;
+
+const project = Project.build();
+
+project.name = "Project Name";
+await project.save();
+```
+
+## Model
+
+A model in TruffleDB is an abstraction representing a namespace in LevelDB. Models are defined as plain JavaScript classes that extend a base `Model` class. The model definition tells Truffle DB the basic details of its represented entity.
+
+A model's class name defines its namespace and its class properties are used to define the JSON object stored in LevelDB. In a relational database, this is equivalent to the table name and table columns.
+
+Model properties can be left undefined, or, provide an object with any combination of `defaultValue`, `required`, or `validation` attributes.
+
+#### `defaultValue`
+
+A newly created instance of the model will set this value for the property.
+
+#### `required`
+
+If this property is `undefined` or `null` the record will not save to the database.
+
+#### `validation`
+
+This function is invoked when the `save` operation is called. It must return `true` or the save operation will error. The validation function will pass its property value to the function as a parameter.
+
+#### `keyField` - TODO
+
+Currently set to 'id' for all models. Used to denote a property on the Model that serves as the key used by the database.
```

Comment:
🎨 Cool to let people change this value. Is there a common reason people do this? Maybe if a key foreign key serves the purpose and you don't need an additional id?

Reply:
Yeah, awesome question. 

There was an ask that we support [Content Addressable Storage IDs](https://en.wikipedia.org/wiki/Content-addressable_storage). It makes a lot of sense for anything that is deployed. I wasn't sure if an internal 'id' field used by the db and an additional CAS id for the user would be helpful - or if the CAS id could do both roles. 

For anything under active development with changing content I think db id + CAS id will help when cycling historical versions as the CAS id would change based on content (especially when multiple devs work off the same project data). And it would make it easier to identify which historical versions made it on chain.   

Your point makes a lot of sense and giving the ability to change the key field may introduce more complexity than is necessary.  Really, with the way sublevel namespaces the db it would be possible to use something like `${projectName}-CASID` as a sort of composite/foreign key solution. Definitely interesting and worth trying or pursuing.

The shorter answer was I just didn't know the correct solution at the time and sided with flexibility until getting some feedback.